### PR TITLE
refactor: move the common part of the heartbeat response handler to common

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Move the common part of the heartbeat response handler to common

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1126

### CODE REVIEW

File: src/common/heartbeat/src/handler/close_region.rs
Comment (MichaelScofield) [inline src/common/heartbeat/src/handler/close_region.rs:1]: I don't think it's a good idea to put `OpenRegionHandler` and `CloseRegionHandler` in heartbeat crate. The two are Datanode only, not "common" at all.

Also we have too many "common" crate, heartbeat's common codes can be placed in `common-meta`. After all, heartbeat is closely related to meta.

**Comment by WenyXu** [comment]
PTAL @MichaelScofield

File: src/common/meta/src/error.rs [hunk 1]
```diff
@@ -13,11 +13,30 @@
...
 #[derive(Debug, Snafu)]
 #[snafu(visibility(pub))]
 pub enum Error {
+    #[snafu(display("Failed to encode object into json, source: {}", source))]
+    EncodeJson {
+        location: Location,
+        source: JsonError,
+    },
+
+    #[snafu(display("Failed to decode object into json, source: {}", source))]
```
Comment (MichaelScofield) [inline src/common/meta/src/error.rs:28]:
```suggestion
    #[snafu(display("Failed to decode object from json, source: {}", source))]
```

**Review by MichaelScofield — APPROVED** [review]

File: src/datanode/src/error.rs [hunk 2]
```diff
@@ -46,18 +38,25 @@ pub enum Error {
         source: catalog::error::Error,
     },
 
+    #[snafu(display("Failed to register table: {}, source: {}", table_name, source))]
+    RegisterTable {
+        table_name: String,
+        #[snafu(backtrace)]
```
Comment (waynexia) [inline src/datanode/src/error.rs:44]:
```suggestion
        location: Location,
```

File: src/datanode/src/error.rs [hunk 3]
```diff
@@ -46,18 +38,25 @@ pub enum Error {
...
         #[snafu(backtrace)]
         source: TableError,
     },
 
-    #[snafu(display("Failed to register table: {}, source: {}", table_name, source))]
-    RegisterTable {
+    #[snafu(display("Failed to get table: {}, source: {}", table_name, source))]
+    GetTable {
         table_name: String,
         #[snafu(backtrace)]
```
Comment (waynexia) [inline src/datanode/src/error.rs:58]:
```suggestion
        location: Location,
```

File: src/datanode/src/error.rs [hunk 4]
```diff
@@ -73,8 +72,24 @@ pub enum Error {
...
+    CheckRegion {
+        table_name: String,
+        #[snafu(backtrace)]
+        source: TableError,
+        region_number: RegionNumber,
+    },
+
+    #[snafu(display("Failed to handle heartbeat response, source: {}", source))]
+    HandleHeartbeatResponse {
+        #[snafu(backtrace)]
```
Comment (waynexia) [inline src/datanode/src/error.rs:90]:
```suggestion
        location: Location,
```

File: src/datanode/src/error.rs [hunk 5]
```diff
@@ -73,8 +72,24 @@ pub enum Error {
...
-    SendMessage { err_msg: String, location: Location },
+    #[snafu(display(
+        "Failed to check region {} in table: {}, source: {}",
+        region_number,
+        table_name,
+        source
+    ))]
+    CheckRegion {
+        table_name: String,
+        #[snafu(backtrace)]
```
Comment (waynexia) [inline src/datanode/src/error.rs:83]:
```suggestion
        location: Location,
```

**Pull request merged**